js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
+      fontColor: '#fff',
+      // available value is 'default', 'border' and 'outside'
+      position: 'default'
+    }
+  };
+}
+
+
+// ********************************************************************
+// * "Class" that creates and manages all charts and the filter panel.
+// *
+// * Constructor arguments:
+// *   chart_configs: array of chart config objects,
+// *                  e.g. [{type: 'bar', group_by:'year'}, ...]
+// *   charts_elt_id: id of HTML element to put charts in
+// *   filters_elt_id: id of HTML element to put filter checkboxes in
+// *   chart_wrapper: HTML for wrapper around chart canvas object
+// *   count_template: HTML for record-count element, containing "{count}"
+// *                   somewhere (which will be replaced with the record count).
+// ********************************************************************
+
+
+var ChartView = function(chart_configs, charts_elt_id, filters_elt_id, chart_wrapper, count_template){
+
+  this.state = {
+    data: null,
+    active_filters: [], //put active filters here
+    charts: [],
+    $count: null
+  }
+
+  this.chart_configs = chart_configs;
+  this.charts_elt_id = charts_elt_id;
+  this.filters_elt_id = filters_elt_id;
+  this.chart_wrapper = chart_wrapper;
+  this.count_template = count_template;
+  this.filters = null;
+
+  this.get_data();
+}
+
+ChartView.prototype.missing_data_label = '(not given)';
+
+ChartView.prototype.get_data = function() {
+  var that = this;
+  jQuery.getJSON('/cleaned_custodial_death_reports.json')
+    .done(function(data){
+      that.state.data = data;
+      that.parse_data();
+      that.create_charts();
+      that.create_filter_panel();
+      that.attach_events();
+    })
+    .fail(function(e){
+      console.log('error fetching data from: ' + this.state.url, e);
+    })
+}
+
+ChartView.prototype.parse_data = function() {
+
+  var that = this;
+  var column_whitelist = [
+    'year', 'race', 'sex', 'manner_of_death', 'age_group',
+    'type_of_custody', 'death_location_type', 'means_of_death',
+  ]
+
+    // Create new columns
+
+  _.each(this.state.data, function(data_row, i) {
+    // Build column for year
+    data_row['year'] = data_row['death_date'].substring(0, 4);
+
+    // Create age group buckets
+    if (data_row['age_at_time_of_death'] < 0) {
+      data_row['age_group'] = null;
+    } else {
+      age_decade = Math.floor(data_row['age_at_time_of_death'] / 10) * 10
+      if (age_decade > 59) {
+        data_row['age_group'] = '60+'
+      } else {
+        data_row['age_group'] = age_decade + '-' + (age_decade + 9)
+      }
+    }
+
+    _.each(column_whitelist, function(key) {
+      if (data_row[key] === undefined || data_row[key] === null || data_row[key] === '') {
+        data_row[key] = that.missing_data_label;
+      }
+    });
+  });
+
+  // Generate filter values
+
+  filters = []
+  _.each(column_whitelist, function(column) {
+    // Get a sorted list of all the unique values for this column
```

Comment:
Being super ⛏-y here.  Can we move the comment outside the loop and update it to:
 // Get a sorted list of all the unique values for each column

Reply:
![ferris_nagging](https://user-images.githubusercontent.com/1421476/41822141-b704c490-77b0-11e8-80a9-7d311d015da1.gif)

Yeah ok